src/Features/LanguageServer/Protocol/Features/Options/SyntaxFormattingOptionsStorage.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Formatting;
+
+internal interface ISyntaxFormattingOptionsStorage : ILanguageService
+{
+    SyntaxFormattingOptions GetOptions(IGlobalOptionService globalOptions);
+}
+
+internal static class SyntaxFormattingOptionsStorage
+{
+    public static ValueTask<SyntaxFormattingOptions> GetSyntaxFormattingOptionsAsync(this Document document, IGlobalOptionService globalOptions, CancellationToken cancellationToken)
```

Comment:
Adds convenience extension methods to read formatting options  from document with fallback to global options.